runtime/libpgmath/lib/common/ceil.c

```diff
@@ -6,8 +6,40 @@
  */
 
 #include "mthdecls.h"
-#if     defined(__SSE4_1__) || defined(__AVX__)
-#include    <immintrin.h>
+
+#if defined(TARGET_X8664)
+/*
+ * For X8664, implement both SSE and AVX versions of __mth_i_ceil using ISA
+ * instruction extensions.
+ *
+ * Using inline assembly allows both the SSE and AVX versions of the routine
+ * to be compiled in a single unit.
+ *
+ * The following asm statements is equivalent to:
+ *      return _mm_cvtss_f32(_mm_ceil_ss(_mm_set1_ps(x), _mm_set1_ps(x)));
+ * But without the need for separate compiliations for SSE4.1 and AVX ISA
+ * extensions.
+ */
+
+float
+__mth_i_ceil_sse(float x)
+{
+  __asm__(
+    "roundss $0x2,%0,%0"
```

Comment:
Is it correct to hardcode the rounding mode here? What if the program has called `ieee_set_rounding_mode`?